sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ExtRefService.java

```diff
@@ -88,12 +93,47 @@ private static List<SclReportItem> checkIedUnityOfIcdSystemVersionUuid(SclRootAd
             .filter(entry -> StringUtils.isNotBlank(entry.getKey()))
             .filter(entry -> entry.getValue().size() > 1)
             .map(entry -> SclReportItem.fatal(entry.getValue().stream()
-                .map(tied -> new IEDAdapter(sclRootAdapter, tied))
-                .map(IEDAdapter::getXPath)
-                .collect(Collectors.joining(", ")),
+                    .map(tied -> new IEDAdapter(sclRootAdapter, tied))
+                    .map(IEDAdapter::getXPath)
+                    .collect(Collectors.joining(", ")),
                 "/IED/Private/compas:ICDHeader[@ICDSystemVersionUUID] must be unique" +
                     " but the same ICDSystemVersionUUID was found on several IED."))
             .collect(Collectors.toList());
     }
 
+    public static SclReport createDataSetAndControlBlocks(SCL scd) {
```

Comment:
Wh 3 methods instead of one method?

Reply:
The issue specifies 3 uses case : 
1.  process all ExtRef in SCD
2. process all ExtRef in an IED (based on iedName)
3. process all ExtRef in a LDevice (based on iedName and ldInst)
I will have to test and maintain these 3 uses cases, even if I write a single method.

If I write one method, I have to document that the method accepts null parameters for iedName and ldInst. I prefer Method Overloading rather than nullable parameters in a public API. This is purely subjective and I am open to discussion.